src/js/builtins/Glob.ts

```diff
@@ -0,0 +1,21 @@
+interface Glob {
+  __scan(opts);
```

Comment:
this is not great, people might try to use it as an API

Reply:
We should make this private somehow, people _will_ use it as an API.

Reply:
We could probably make it private with the `#` syntax so we have `#__scan` which is only accessible within the class

Reply:
We can use private symbols for this. `vm.commonIdentifiers.match` and `vm.commonIdentifiers.test` I think will work.